src/Zoo.cpp

```diff
@@ -0,0 +1,486 @@
+				if (found)
+				{
+					std::cout << "Enter name of animal: ";
+					std::cin >> name;
+					create(type, name);
+				}
+				else
+				{
+					throw (type);
+				}
+			}
+			catch (const std::string type)
+			{
+				std::cout << "Invalid " << type << " type, try again!\n\n";
+			}
+			step();
+			printListAndMap();
+			break;
+		case 4:
+			try
+			{
+				found = false;
+				std::cout << "Enter number of animal: ";
+				std::cin >> numOfAnimal;
+				std::string number = std::to_string(numOfAnimal);
+				for (int i = 0; i < lst.size(); i++)
+				{
+					if (lst[i][0] == number)
+					{
+						found = true;
+						del(numOfAnimal);
+						break;
+					}
+				}
+				if (!found)
+					throw numOfAnimal;
+			}
+			catch (const int number)
+			{
+				std::cout << "Animal number " << number << " does not exist!\n\n";
+			}
+			step();
+			printListAndMap();
+			break;
+		case 5:
+			try
+			{
+				std::cout << "Enter type of animal: ";
+				std::cin >> type;
+				found = (std::find(classes.begin(), classes.end(), type) != classes.end());
+				if (found)
+				{
+					delAll(type);
+				}
+				else
+				{
+					throw (type);
+				}
+			}
+			catch (const std::string type)
+			{
+				std::cout << "Invalid " << type << " type, try again!\n\n";
+			}
+			step();
+			printListAndMap();
+			break;
+		case 6:
+			help();
+			printListAndMap();
+			break;
+		case 7:
+			step();
+			printListAndMap();
+			break;
+
+		default:
+			std::cout << "The command does not exist, try again!\n";
+			break;
+		}
+
+	} while (choice != 0);
+}
+/* Accepts as a parameter the number of the animal.
+   This animal will not move from here on out, unless the move command is applied to it again */
+void Zoo::stop(int numOfAnimal) const
+{
+	Animal* animal;
+	std::string animalNum = std::to_string(numOfAnimal);
+	int index;
+
+	for (int i = 0; i < lst.size(); i++)
+	{
+		if (lst[i][0] == animalNum)
+		{
+			index = numOfAnimal - 1;
+			animal = animalList[index];
+			animal->stop();
+			break;
+
+		}
+	}
+}
+
+/* Accepts as a parameter the number of the animal.
+   The command has meaning only if the animal has stopped before. When the animal starts to move, it resets its direction.*/
+void Zoo::move(int numOfAnimal) const
+{
+
+	Animal* animal;
+	std::string animalNum = std::to_string(numOfAnimal);
+	int index;
+
+	for (int i = 0; i < lst.size(); i++)
+	{
+		if (lst[i][0] == animalNum)
+		{
+			index = numOfAnimal - 1;
+			animal = animalList[index];
+			animal->move();
+			break;
+		}
+	}
+}
+
+/* Creating another animal. Accepts as parameters the type of animal and a name for it.
+   The location of the animal will be randomly drawn.*/
+void Zoo::create(std::string  name, std::string type)
+{
+
+	Animal::Location loc;
+	Animal* animal;
+
+
+	int rows = 20;
+	int columns = 40;
+
+	int row = rand() % rows;
+	int column = rand() % columns;
+
+	std::string location = std::to_string(row) + "," + std::to_string(column);
```

Comment:
you can use operator overloading to print the location why do it manualy ??
define operator<< to print this

Reply:
I fixed it.